### DISCUSSION TITLE
RuntimeError: applying transform  on LoadImaged

### DISCUSSION THREAD

**Discussion by lg2578** [Project-MONAI/tutorials#1312] [Q&A]
code:
```
import monai
import torch
from monai import transforms 
a={'image': 'd:/datas/Task801_stomach+bowel/imagesTr/case101_day22_image.nii', 'label': 'd:/datas/Task801_stomach+bowel/labelsTr/case101_day22_mask_multiclass.nii'}
# b="/Users/lezi/downloads/case154day14_121_0000.nii.gz"
# b1="/Users/lezi/downloads/case119day0_033.nii.gz"
# a={'image':b,'label':b1}
T=transforms.LoadImaged(keys=['image','label'])
T2=transforms.LoadImage()
a=T2(a['image'])
a2=T(a)
```
error message:
InvalidDicomError: File is missing DICOM File Meta Information header or the 'DICM' prefix is missing from the header. Use force=True to force reading.
my env:
monai 1.11,pydicom 2.3.1,pytorch 2.0 ,python 3.11
when in another pc with monai 1.11, python 3.9.2 ，this error doesn't happen.

**Comment by KumoLiu** [comment]
Hi @lg2578, could you please try `NibabelReader` seems your data is ending up with `nii`.
```
LoadImaged(keys=['image','label'], reader="NibabelReader")
```
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L844
https://github.com/Project-MONAI/MONAI/blob/b356fecdb265ee9af4ce91a0c8238731994b5095/monai/transforms/io/array.py#L147-L153

Hope it can help you, thanks!

**Reply by lg2578** [reply]
File [c:\Users\lezi\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydicom\filereader.py:1021](file:///C:/Users/lezi/AppData/Local/Programs/Python/Python311/Lib/site-packages/pydicom/filereader.py:1021), in dcmread(fp, defer_size, stop_before_pixels, force, specific_tags)
   1019     stop_when = _at_pixel_data
   1020 try:
...
    727     )
    728 else:
    729     logger.debug(f"{fp.tell() - 4:08x}: 'DICM' prefix found")

InvalidDicomError: File is missing DICOM File Meta Information header or the 'DICM' prefix is missing from the header. Use force=True to force reading.

****************
RuntimeError: LoadImage cannot find a suitable reader for file: d:\datas\Task801_stomach+bowel\imagesTr\case101_day22_image.nii.
    Please install the reader libraries, see also the installation instructions:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies.
   The current registered: [, , ].
Traceback (most recent call last):
...
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[c:\Users\lezi\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydicom\filereader.py](file:///C:/Users/lezi/AppData/Local/Programs/Python/Python311/Lib/site-packages/pydicom/filereader.py)", line 723, in read_preamble
    raise InvalidDicomError(
pydicom.errors.InvalidDicomError: File is missing DICOM File Meta Information header or the 'DICM' prefix is missing from the header. Use force=True to force reading.
*******************
my code:
import monai
import torch
from monai import transforms 

a={'image': 'd:/datas/Task801_stomach+bowel/imagesTr/case101_day22_image.nii', 'label': 'd:/datas/Task801_stomach+bowel/labelsTr/case101_day22_mask_multiclass.nii'}
#b="/Users/lezi/downloads/case154day14_121_0000.nii.gz"
#b1="/Users/lezi/downloads/case119day0_033.nii.gz"
#a={'image':b,'label':b1}
T=transforms.LoadImaged(keys=['image','label'], reader="NibabelReader")
T2=transforms.LoadImage(reader="NibabelReader")
a=T2(a['image'])
a2=T(a)

**Reply by lg2578** [reply]
import nibabel as nib
from monai import transforms 

a={'image': 'd:/datas/Task801_stomach+bowel/imagesTr/case101_day22_image.nii', 'label': 'd:/datas/Task801_stomach+bowel/labelsTr/case101_day22_mask_multiclass.nii'}
T=transforms.LoadImaged(keys=['image','label'], reader="NibabelReader")
T2=transforms.LoadImage(reader="NibabelReader")
nib.load(a['image']).get_fdata().shape
**********
the above code works

**Reply by lg2578** [reply]
maybe it just caused by no  nibabel package under python 3.11 env

**Reply by lg2578** [reply]
or by the code:
a=T2(a['image'])
a2=T(a)
a is a image not a file in second line.

**Reply by KumoLiu** [reply]
Hi @lg2578, I think you are right.
For `LoadImage`, the input should be "path file or file-like object or a list of files"
https://github.com/Project-MONAI/MONAI/blob/b356fecdb265ee9af4ce91a0c8238731994b5095/monai/transforms/io/array.py#L245-L250
And also for `LoadImaged`, the input should still be a path file instead of the images.

Hope it can help you, thanks!
